docarray/index/backends/elastic.py

```diff
@@ -141,13 +130,19 @@ def find(
             else:
                 query_vec = query
             query_vec_np = BaseDocIndex._to_numpy(self._outer_instance, query_vec)
-            self._query['size'] = limit
-            self._query['query']['script_score'] = ElasticV7DocIndex._form_search_body(
-                query_vec_np, limit, search_field
-            )['query']['script_score']
+            self._query['knn'] = ElasticDocIndex._form_search_body(
+                query_vec_np,
+                limit,
+                search_field,
+                self._outer_instance._runtime_config.default_column_config[
+                    'dense_vector'
+                ]['num_candidates'],
+            )['knn']
 
             return self
 
+        # filter accrpts Leaf/Compound query clauses
+        # https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl.html
```

Comment:
I cannot comment below, but in the `raise_not_composabe()` calls, the arguments that are passed should be "find_batched", "filter_batched", and "text_search_batched", not "find_batched" x2 and "text_search"